cli/receivers.go

```diff
@@ -0,0 +1,308 @@
+// Copyright 2022 Prometheus Team
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"context"
+	"errors"
+	"sort"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/alertmanager/config"
+	"github.com/prometheus/alertmanager/notify"
+	"github.com/prometheus/alertmanager/notify/discord"
+	"github.com/prometheus/alertmanager/notify/email"
+	"github.com/prometheus/alertmanager/notify/opsgenie"
+	"github.com/prometheus/alertmanager/notify/pagerduty"
+	"github.com/prometheus/alertmanager/notify/pushover"
+	"github.com/prometheus/alertmanager/notify/slack"
+	"github.com/prometheus/alertmanager/notify/sns"
+	"github.com/prometheus/alertmanager/notify/telegram"
+	"github.com/prometheus/alertmanager/notify/victorops"
+	"github.com/prometheus/alertmanager/notify/webhook"
+	"github.com/prometheus/alertmanager/notify/wechat"
+	"github.com/prometheus/alertmanager/template"
+	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/common/promlog"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	maxTestReceiversWorkers = 10
+)
+
+var (
+	ErrNoReceivers = errors.New("no receivers with configuration set")
+)
+
+type TestReceiversParams struct {
+	Alert     *TestReceiversAlertParams
+	Receivers []*config.Receiver
+}
```

Comment:
Curious as to thoughts on if we can also accept custom templates here?

If the command could accept extra template files, it'd significantly increase the usefulness IMO, as users can test and iterate on their formatted message in addition to only validating the receiver config.

In the latest main, there's an easy way to stitch in additional template files into an already constructed `template.Template` instance - so making test files available in the receivers would be a straightforward extension after calling `getTemplate()` later on.